src/Tools/ExternalAccess/FSharp/Editor/IFSharpEditorInlineRenameService.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Editor
+{
+    internal struct FSharpInlineRenameLocation
+    {
+        public Document Document { get; }
+        public TextSpan TextSpan { get; }
+
+        public FSharpInlineRenameLocation(Document document, TextSpan textSpan) : this()
+        {
+            this.Document = document;
+            this.TextSpan = textSpan;
+        }
+    }
+
+    internal enum FSharpInlineRenameReplacementKind
+    {
+        NoConflict,
+        ResolvedReferenceConflict,
+        ResolvedNonReferenceConflict,
+        UnresolvedConflict,
+        Complexified,
+    }
+
+    internal struct FSharpInlineRenameReplacement
+    {
+        public FSharpInlineRenameReplacementKind Kind { get; }
+        public TextSpan OriginalSpan { get; }
+        public TextSpan NewSpan { get; }
+
+        public FSharpInlineRenameReplacement(FSharpInlineRenameReplacementKind kind, TextSpan originalSpan, TextSpan newSpan) : this()
```

Comment:
```suggestion
        public FSharpInlineRenameReplacement(FSharpInlineRenameReplacementKind kind, TextSpan originalSpan, TextSpan newSpan)
```

Reply:
Makes sense to me. Also note that the Roslyn `InlineRenameReplacement` has `this()`. It's obvious now that I copied and pasted it.